### PR TITLE
feat: Move Driver config to new custom config section

### LINKED ISSUE #250: Move Driver config to new custom config section
The Driver configuration can benefit to being moved to a structure custom configuration section which eliminates the need to convert the string map to the existing config structure. 

Need to determine if any of the configuration needs to be writable. If so make a sub section of the custom config writable by  watching for changes.

### PR DESCRIPTION
**Pull request by weichou1229**
Close #250

Signed-off-by: weichou <weichou1229@gmail.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-mqtt-go/blob/master/.github/CONTRIBUTING.md

## What is the new behavior?
- Update configuration.toml
- Load config from custom config section

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

### CODE REVIEW

File: cmd/res/configuration.toml [hunk 1]
```diff
@@ -89,25 +89,27 @@ EnableAsyncReadings = true
 AsyncBufferSize = 16
 UseMessageBus = false
 
-# Driver configs
-[Driver]
+# Custom driver configs
+[CustomConfig]
```
Comment (cloudxxx8) [inline cmd/res/configuration.toml:93]: rename to MQTTBrokerInfo

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]

File: cmd/res/configuration.toml (hunk 1 above)
Comment (weichou1229) [reply cmd/res/configuration.toml]: Renamed.

File: internal/driver/driver.go [hunk 2]
```diff
@@ -62,12 +45,15 @@ func NewProtocolDriver() sdkModel.ProtocolDriver {
...
-	if err != nil {
-		return fmt.Errorf("read MQTT driver configuration failed: %w", err)
+	ds := service.RunningService()
+
+	if err := ds.LoadCustomConfig(d.serviceConfig, CustomConfigSectionName); err != nil {
+		return fmt.Errorf("unable to load '%s' custom configuration: %s", CustomConfigSectionName, err.Error())
 	}
-	d.Config = config
+
+	lc.Infof("Custom config is: %v", d.serviceConfig)
```
Comment (cloudxxx8) [inline internal/driver/driver.go:56]: This should be the debug log

Comment (weichou1229) [reply internal/driver/driver.go]: Modified.

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**